core/com.b2international.snowowl.core/src/com/b2international/snowowl/core/identity/JWTSupport.java

```diff
@@ -0,0 +1,281 @@
+	private static final JWTVerifier JWT_VERIFIER_DISABLED = new JWTVerifier() {
+		@Override
+		public DecodedJWT verify(DecodedJWT arg0) throws JWTVerificationException {
+			throw new BadRequestException("JWT token verification is not configured.");
+		}
+		
+		@Override
+		public DecodedJWT verify(String arg0) throws JWTVerificationException {
+			throw new BadRequestException("JWT token verification is not configured.");
+		}
+	};
+
+	// Disabled JWT Generator implementation that throws an exception if any token signing related logic would be required
+	private static final JWTGenerator JWT_GENERATOR_DISABLED = new JWTGenerator() {
+		@Override
+		public String generate(User user) {
+			throw new BadRequestException("JWT token signing is not configured.");
+		}
+		
+		@Override
+		public String generate(String email, Map<String, Object> claims) {
+			throw new BadRequestException("JWT token signing is not configured.");
+		}
+	};
+	
+	private final JWTConfiguration config;
+	
+	private JWTGenerator generator;
+	private JWTVerifier verifier;
+	
+	JWTSupport(JWTConfiguration config) {
+		this.config = config;
+	}
+	
+	public void init() throws Exception {
+		RSAKeyProvider rsaKeyProvider = createRSAKeyProvider(config);
+		Algorithm algorithm;
+		if (!Strings.isNullOrEmpty(config.getJws())) {
+			algorithm = SUPPORTED_JWS_ALGORITHMS.getOrDefault(config.getJws(), JWTSupport::throwUnsupportedJws).apply(config, rsaKeyProvider);
+		} else {
+			IdentityProvider.LOG.warn("'identity.jws' configuration is missing, disabling JWT authorization token signing and verification.");
+			algorithm = null;
+		}
+		
+		if (algorithm == null) {
+			// both signing and verification is disabled
+			generator = JWT_GENERATOR_DISABLED;
+			verifier = JWT_VERIFIER_DISABLED;
+		} else if (rsaKeyProvider != null && rsaKeyProvider.getPrivateKey() == null) {
+			generator = JWT_GENERATOR_DISABLED;
+			verifier = createJWTVerifier(algorithm, config);
+		} else {
+			generator = new DefaultJWTGenerator(algorithm, config.getIssuer(), config.getEmailClaimProperty(), config.getPermissionsClaimProperty());
+			verifier = createJWTVerifier(algorithm, config);
+		}
+	}
+	
+	@VisibleForTesting
+	/*package*/ DecodedJWT verify(String token) {
+		return verifier.verify(token);
+	}
+	
+	public User authJWT(String token) {
+		return toUser(verify(token), config);
+	}
+	
+	/**
+	 * Converts the given JWT access token to a {@link User} representation using the configured email and permission claims. This method does not
+	 * verify the given access token, it only decodes it and uses the publicly available claims to construct the {@link User} object. To verify a
+	 * token and create a user object use the {@link #authJWT(String)} method.
+	 * 
+	 * @param token
+	 * @return a {@link User} instance created from the given token
+	 */
+	public User toUser(String token) {
+		return toUser(JWT.decode(token), config);
+	}
```

Comment:
Can you double check that all current callers use these two methods as intended (ie. no one takes JWTs at face value unless they were verified at an earlier point in time)?

Reply:
The second method is not used at all, we can safely remove it.